### ISSUE TITLE
Which library variants require asynchronous initialization?

### ISSUE THREAD

**Issue by DanielSWolf** [mmig/libflac.js#32]
All samples contain this placeholder comment:

> //NOTE if async-library variant is used, should wait for initialization:
> //Flac.onready(function(){ ...

In [the documentation](https://mmig.github.io/libflac.js/doc/#async-initialization-with-webpack), the chapter "Async Initialization" states that "Some variants of the `libflac.js` library are loaded asynchronously". But I can't find the information which variants these are exactly.

Which variants require asynchronous initialization, which don't?

**Comment by russaa** [comment]
generally, the asnychonous initialization is relevant for all library variants that have additional, required library files (i.e. in addition to the main JavaScript library file)

the section [Library Variants](https://github.com/mmig/libflac.js#library-variants) (or specifically the sub-sections [Default Library](https://github.com/mmig/libflac.js#default-library), [Minified Library](https://github.com/mmig/libflac.js#minified-library), and [Development Library](https://github.com/mmig/libflac.js#development-library)) lists the additional required files per variant:

in short:
 * all `WASM` variants, as well as the _minified_ variant are loaded asynchronously
 * the `asm.js` variants for _default_ and _development_ are initialized sychronously

**Comment by DanielSWolf** [comment]
Thanks for the clarification!
